src/autopas/tuning/AutoTuner.cpp

```diff
@@ -161,7 +163,7 @@ std::tuple<Configuration, bool> AutoTuner::getNextConfig() {
   // If we are not (yet) tuning or there is nothing to tune return immediately.
   if (not inTuningPhase()) {
     return {getCurrentConfig(), false};
-  } else if (getCurrentNumSamples() < _maxSamples) {
+  } else if (getCurrentNumSamples() < _maxSamples and getCurrentNumSamples() > 0) {
```

Comment:
That's a slight leftover from something else I had tried.
The idea is that the number of current samples is never 0, except for the very first iteration. 
Previously, we initialized `_samplesNotRebuildingNeighborLists` to contain `_maxSamples` dummy samples, such that we start in the `else` clause here. I found it cleaner to initialize `_samplesNotRebuildingNeighborLists` to be empty, which then needs this extra condition to catch the else in the very first iteration.

Reply:
Hmm, if you feel that it's better, I won't stand in your way. 
I just think it is a bit more complex because this way you are treating the case that we start the very first tuning phase vs any other tuning phase differently (empty `_samplesNotRebuildingNeighborLists` vs full `_samplesNotRebuildingNeighborLists`), effectively, when modeling the AutoTuner as a state machine, introducing a new state. One goal of my last rewrite of the AutoTuner was to reduce the number of states.

Reply:
Nah, you are right, that's an unnecessary change. Just something that confused me at some point, but this solution is not clearly better. So I just reverted it to the old behavior in [6f1403a](https://github.com/AutoPas/AutoPas/pull/751/commits/6f1403a7925df894f8277b8f878d30a440e46c19)